wordpress-mobile/WordPress-FluxC-Android
# Add XPostsStore for Retrieving a site's xpost candidates

Related [WPAndroid PR](https://github.com/wordpress-mobile/WordPress-Android/pull/13184)

This adds the capability for retrieving and persisting a user/sites xposts. 

I decided to persist this data in two tables to avoid duplicating site information when the same site was an xpost candidate for multiple sites. One table simply persists the site information for a given xpost candidate (`XpostSite`) (name, URL, subdomain, blavatar, etc.). I could have used the `SiteModel` for this, but updating that object with information from another endpoint felt a bit risky with little (if any) benefit, so I kept them separate. I then added a second table that serves the purpose of connecting source site (`SiteModel`), to the relevant xpost target sites (`XPostSite`). That table makes it possible to look up all the `XPostSites` available for a given `SiteModel`.

### Testing

This can be tested via the related [WPAndroid PR](https://github.com/wordpress-mobile/WordPress-Android/pull/13184). In addition, it can be tested in the example app:

1. Log into a P2-themed site in the example app
2. Tap on "Sites"
3. Tap on "Fetch XPosts"
4. Observe that a successful response is displayed in the log at the bottom of the screen.
